pkg/server/api/agent/v1/service.go

```diff
@@ -275,8 +410,16 @@ func (s *Service) RenewAgent(stream agent.Agent_RenewAgentServer) error {
 		log.Errorf("Invalid argument: expected params step but got %T", params)
 		return status.Errorf(codes.InvalidArgument, "expected params step but got %T", params)
 	}
+	if params.Params == nil {
+		log.Error("Invalid argument: missing Params")
+		return status.Error(codes.InvalidArgument, "missing Params")
+	}
+	if len(params.Params.Csr) == 0 {
```

Comment:
@MarcosDY I moved the validation outside of the sign function, and added an extra one. I also changed the signature slightly to make it usable in other places. 
Let me know if that is OK.